### PR TITLE
Update deprecated api versions

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: config/common/operator/clusterrole-operator.yaml [hunk 1]
```diff
@@ -31,3 +31,20 @@ rules:
       - get
       - update
       - delete
+  - apiGroups:
```
Comment (0sewa0) [inline config/common/operator/clusterrole-operator.yaml:34]: webhook stuff is not necessary

File: Makefile [hunk 2]
```diff
@@ -18,12 +18,13 @@ ifeq ($(origin IMG),undefined)
...
 else
-IMG=quay.io/dynatrace/dynatrace-operator:snapshot-$(shell git branch --show-current | sed "s\#[^a-zA-Z0-9_-]\#-\#g")
+IMG=quay.io/dynatrace/dynatrace-operator:snapshot-$(shell git branch --show-current | sed "s/[^a-zA-Z0-9_-]/-/g")
 endif
 endif
 
 # Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:trivialVersions=true"
+CRD_OPTIONS ?= "crd:trivialVersions=true, preserveUnknownFields=false, crdVersions=v1"
+CRD_OPTIONS_OCP311 ?= "crd:trivialVersions=true, preserveUnknownFields=false, crdVersions=v1beta1"
```
Comment (baichinger) [inline Makefile:27]: That blanks are adventurous. This string must be treated as one argument to controller-gen.

File: Makefile [hunk 3]
```diff
@@ -1,9 +1,9 @@
+SHELL = bash
+
 # Current Operator version
 VERSION ?= 0.0.1
 # Default bundle image tag
 BUNDLE_IMG ?= controller-bundle:$(VERSION)
-# Default platform for bundle
-PLATFORM="kubernetes"
```
Comment (baichinger) [inline Makefile:6]: This variable is still used in target bundle.

File: config/common/webhook/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (0sewa0) [inline config/common/webhook/kustomization.yaml:1]: more webhook stuff

File: config/common/webhook/mutatingwebhookconfiguration.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-apiVersion: admissionregistration.k8s.io/v1beta1
```
Comment (0sewa0) [inline config/common/webhook/mutatingwebhookconfiguration.yaml:1]: webhook stuff

File: config/openshift3.11/operator/deployment-operator-patch.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+- op: add
+  path: /spec/template/spec/containers/0/env/-
+  value:
+    name: RELATED_IMAGE_DYNATRACE_ONEAGENT
+    value: registry.connect.redhat.com/dynatrace/oneagent
+- op: add
+  path: /spec/template/spec/containers/0/env/-
+  value:
+    name: DISABLE_WEBHOOK
```
Comment (0sewa0) [inline config/openshift3.11/operator/deployment-operator-patch.yaml:9]: webhook stuff

**Review by baichinger — CHANGES_REQUESTED** [review]
Should this work with the latest version of operator-sdk as well or do I need a specific one? With v1.10 it fails:
```
% make bundle             
controller-gen "crd:trivialVersions=true, preserveUnknownFields=false, crdVersions=v1" paths="./..." output:crd:artifacts:config=config/crd/default/bases
operator-sdk generate kustomize manifests -q
WARN[0000] Config version 3-alpha has been stabilized as 3, and 3-alpha is no longer supported. Run `operator-sdk alpha config-3alpha-to-3` to upgrade your PROJECT config file to version 3 
Error: error reading configuration: unable to load the configuration: unable to create config for version "3-alpha": version 3-alpha is not supported
```

File: Makefile (hunk 2 above)
Comment (DTMad) [reply Makefile:27]: Removed the blanks

File: config/common/operator/clusterrole-operator.yaml (hunk 1 above)
Comment (DTMad) [reply config/common/operator/clusterrole-operator.yaml:34]: Removed

File: config/common/webhook/kustomization.yaml (hunk 4 above)
Comment (DTMad) [reply config/common/webhook/kustomization.yaml:1]: Removed the whole folder

File: config/common/webhook/mutatingwebhookconfiguration.yaml (hunk 5 above)
Comment (DTMad) [reply config/common/webhook/mutatingwebhookconfiguration.yaml:1]: removed the whole folder

File: config/openshift3.11/operator/deployment-operator-patch.yaml (hunk 6 above)
Comment (DTMad) [reply config/openshift3.11/operator/deployment-operator-patch.yaml:9]: Removed

File: Makefile (hunk 3 above)
Comment (DTMad) [reply Makefile:6]: Re-added

**Review by gkrenn — APPROVED** [review]

**Comment by DTMad** [comment]
> Should this work with the latest version of operator-sdk as well or do I need a specific one? With v1.10 it fails:
> 
> ```
> % make bundle             
> controller-gen "crd:trivialVersions=true, preserveUnknownFields=false, crdVersions=v1" paths="./..." output:crd:artifacts:config=config/crd/default/bases
> operator-sdk generate kustomize manifests -q
> WARN[0000] Config version 3-alpha has been stabilized as 3, and 3-alpha is no longer supported. Run `operator-sdk alpha config-3alpha-to-3` to upgrade your PROJECT config file to version 3 
> Error: error reading configuration: unable to load the configuration: unable to create config for version "3-alpha": version 3-alpha is not supported
> ```

Fixed

**Review by baichinger — APPROVED** [review]

**Pull request merged**